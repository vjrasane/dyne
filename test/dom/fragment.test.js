import Dyne from "../../src";

describe("fragment", () => {
  it("creates empty fragment", () => {
    const dom = Dyne.Fragment();
    expect(dom).toMatchSnapshot();
  });

  it("creates basic fragment", () => {
    const dom = Dyne.Fragment(null, ["Hello World!", <div>child</div>]);
    expect(dom).toMatchSnapshot();
  });

  it("creates dynamic fragment", () => {
    const Component = (props, children) =>
      Dyne.Fragment(null, [<div {...props}></div>, <div>{children}</div>]);
    const dom = <Component prop="value">child</Component>;
    expect(dom).toMatchSnapshot();
  });
});
