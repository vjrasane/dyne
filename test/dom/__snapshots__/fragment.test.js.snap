// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment creates basic fragment 1`] = `
Array [
  "Hello World!",
  Object {
    "children": Array [
      "child",
    ],
    "props": Object {},
    "type": "div",
  },
]
`;

exports[`fragment creates dynamic fragment 1`] = `
Array [
  Object {
    "children": Array [],
    "props": Object {
      "prop": "value",
    },
    "type": "div",
  },
  Object {
    "children": Array [
      "child",
    ],
    "props": Object {},
    "type": "div",
  },
]
`;

exports[`fragment creates empty fragment 1`] = `Array []`;
